src/service/profile.js

```diff
@@ -101,26 +101,23 @@ export const setUnFollow = async (data) =>
       });
   });
 
+export const followUser = async (data) =>
+  new Promise((resolve, reject) => {
+    api
+      .post('/profiles/follow-user-v3', data)
+      .then((res) => {
+        resolve(res.data);
+      })
+      .catch((err) => {
+        reject(err);
+      });
+  });
+
 export const setFollow = async (data) => {
-  const textTargetUser = `${data.username_follower} started following you. Send them a message now`;
-  const textOwnUser = `You started following ${data.username_followed}. Send them a message now.`;
-  const chat = await createChannel(
-    'messaging',
-    [data.user_id_followed, data.user_id_follower],
-    `${data.username_followed},${data.username_follower}`
-  );
-  chat.update(
-    {
-      name: `${data.username_followed},${data.username_follower}`
-    },
-    {
-      text: textTargetUser,
-      system_user: data.user_id_follower,
-      is_from_prepopulated: true,
-      other_text: textOwnUser
-    },
-    {skip_push: true}
-  );
+  const textTargetUser = `${data.username_follower} started following you.\nSend them a message now`;
+  const textOwnUser = `You started following ${data.username_followed}.\nSend them a message now.`;
+  const members = [data.user_id_follower, data.user_id_followed];
+  await followClient(members, data, textTargetUser, textOwnUser);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `setFollow` function has been significantly refactored. The old version was creating a chat channel and updating it with some information, while the new version is calling `followClient` with similar parameters. However, there's no error handling for the `followClient` call. If this function fails, it could lead to inconsistent state between the follow action and the chat system. Consider adding error handling around this call.
```diff
 export const setFollow = async (data) => {
  const textTargetUser = `${data.username_follower} started following you.\nSend them a message now`;
  const textOwnUser = `You started following ${data.username_followed}.\nSend them a message now.`;
  const members = [data.user_id_follower, data.user_id_followed];
  try {
    await followClient(members, data, textTargetUser, textOwnUser);
  } catch (error) {
    console.error('Error in followClient:', error);
    throw error; // or handle the error as needed
  }
  return new Promise((resolve, reject) => {
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->